### PR TITLE
docs: replica rebuild

### PR DESCRIPTION
**Pull request by AJDatacore**
Added replica rebuild history

### CODE REVIEW

File: reference/replica-rebuild.md [hunk 1]
```diff
@@ -1,10 +1,82 @@
...
 However, the drawback to the above approach was that even if a replica was inaccessible for a short period (e.g., due to a node restart), a full rebuild was triggered. This may not have a significant impact on replicas with small sizes, but it is not desirable for large replicas.
 
 The partial rebuild feature, overcomes the above problem and helps in achieving faster rebuild times. When volume target encounters IO error on a child/replica, it marks the child as `Faulted` (removing it from the I/O path) and begins to maintain a write log for all subsequent writes. The Core agent starts a default 10 minute wait for the replica to come back. If the child's replica is online again within timeout, the control-plane requests the volume target to online the child and add it to the IO path along with a partial rebuild process using the aforementioned write log.
+-->
+
+**Full rebuild** and **partial rebuild** are two approaches used in the context of replica redundancy and data recovery in a distributed storage system.
+
+### Full rebuild
+
```
Comment (abhilashshetty04) [inline reference/replica-rebuild.md]: Full and Partial rebuild strategy is already explained in this page itself right?

File: reference/replica-rebuild.md [hunk 2]
```diff
@@ -1,10 +1,82 @@
...
 
 {% hint style="info %}
 The control plane waits for 10 minutes before initiating the full rebuild process, as the `--faulted-child-wait-period` is set to 10 minutes. To configure this parameter, edit values.yaml.
-{% endhint %}
\ No newline at end of file
+{% endhint %}
+
+## Replica rebuild history 
+
+The data-plane handles both full and partial replica rebuilds. To view the logs of these rebuilds, you can use either the `kubectl` command or make an API call.
```
Comment (dsharma-dc) [inline reference/replica-rebuild.md]: `To view history of the rebuilds that an existing volume target has undergone during its lifecycle till now...`

File: reference/snapshot.md
Comment (dsharma-dc) [inline reference/snapshot.md:1]: Is this snapshot related file supposed to be part of this PR?

File: reference/replica-rebuild.md [hunk 3]
```diff
@@ -1,10 +1,82 @@
...
+
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+kubectl dummy
+```
+{% endtab %}
+
+{% tab title="CURL" %}
```
Comment (tiagolobocastro) [inline reference/replica-rebuild.md]: I don't think this is needed via curl.. otherwise we'd have to document every kubectl command via curl which doesn't make much sense imho.

File: reference/replica-rebuild.md (hunk 2 above)
Comment (tiagolobocastro) [inline reference/replica-rebuild.md]: Are we sure we want to document this as it stands?
We have limitations that this information is not persistent, and will disappear after crash, restart, target being recreated or volume unpublished for whatever reason.
At the very least we need to highlight this..

File: reference/replica-rebuild.md (hunk 1 above)
Comment (AJDatacore) [reply reference/replica-rebuild.md]: Yes, I will retain the old content.

File: reference/replica-rebuild.md (hunk 2 above)
Comment (AJDatacore) [reply reference/replica-rebuild.md]: @abhilashshetty04 I would like your input to resolve this comment.

Comment (abhilashshetty04) [reply reference/replica-rebuild.md]: @AJDatacore , We discussed this on the call :) . Rebuild history information is coupled with volume target. If for any reason volume gets a new target older rebuild history details will not be available as its not persisted. So it would be better if we highlight this limitation.

Comment (AJDatacore) [reply reference/replica-rebuild.md]: Line 86 is a note that highlight itself. 
Shall I move it to the top? @abhilashshetty04

Comment (abhilashshetty04) [reply reference/replica-rebuild.md]: @tiagolobocastro , https://github.com/openebs/mayastor-docs/pull/139/files#diff-4994b61686d97c7166538df1c18705d101b161101d4a99d2b98b56ccdaf6e938R86

This highlights the limitation. Is this okay?

Comment (tiagolobocastro) [reply reference/replica-rebuild.md]: Seems ok to me

**Comment by abhilashshetty04** [comment]
@AJDatacore , Lets remove curl documentation. kubectl should be enough.

File: reference/snapshot.md
Comment (abhilashshetty04) [reply reference/snapshot.md:1]: @AJDatacore removed snapshot commits now.

**Review by abhilashshetty04 — APPROVED** [review]

**Review by avishnu — APPROVED** [review]
lgtm

**Review by tiagolobocastro — APPROVED** [review]

**Comment by AJDatacore** [comment]
Addressed all the comments. Ready for merge.
